### PR TITLE
Refactor Flake

### PR DESCRIPTION
**Pull request by JonathanLorimer**

### CODE REVIEW

**Comment by masaeedu** [comment]
Can you separate the commit that does all the reformatting from the ones that actually add/modify stuff?

**Comment by JonathanLorimer** [comment]
@masaeedu I can't, cause I had to just copy all the files

**Comment by masaeedu** [comment]
@JonathanLorimer Not quite following what that means, do you not remember the changes you made? Or were there a lot of those?

File: flake.nix [hunk 1]
```diff
@@ -9,32 +10,83 @@
...
+          };
+          cofree-bot = hsPkgs.cofree-bot;
+        };
+
+        checks = {
+          pre-commit-check = pre-commit-hooks.lib.${system}.run {
+            src = ./.;
+            hooks = {
+              nixpkgs-fmt.enable = true;
+              # brittany = {
```
Comment (masaeedu) [inline flake.nix]: Should this be removed or restored?

File: src/CofreeBot/Bot/Calculator.hs [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-import CofreeBot.Utils
-import Data.Profunctor
-import Data.Text qualified as T
-import Data.Functor
+import           CofreeBot.Bot
+import           CofreeBot.Bot.Calculator.Language
+import           CofreeBot.Utils
+import           Data.Functor
+import           Data.Profunctor
+import qualified Data.Text                     as T
```
Comment (solomon-b) [inline src/CofreeBot/Bot/Calculator.hs:8]: Is it possible to keep `ImportQualifiedPost`? I don't like having the huge indents on the non qualified imports to maintain alignment.

**Comment by solomon-b** [comment]
Thanks for updating the flakes! I'm not 100% happy with the formatter but i'm ok with committing to it for the sake of consistency. How do i run it on the entire codebase?

**Comment by masaeedu** [comment]
@solomon-b I think `brittany --write-mode=inplace src/**/*.hs`, although we should probably stick that in a script or something. Currently not all files are formatted btw, the merge with main introduced some unformatted stuff.

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,4 @@
 /flake.lock
 /.envrc
 /dist-newstyle/
+.pre-commit-config.yaml
```
Comment (masaeedu) [inline .gitignore:5]: Is this file created on the fly?

**Comment by masaeedu** [comment]
Ok I've split up the commits so that the actual reformatting is separate from the meat of the PR. If something else goes into main before this does, rebase interactively on main and drop the reformat commit (no conflicts due to reformatting), then just reformat and commit.

File: .gitignore (hunk 3 above)
Comment (JonathanLorimer) [reply .gitignore:5]: yeah, its created when you enter the nix shell

**Comment by JonathanLorimer** [comment]
Added a script to allow you to format all files, although you should be able to set your `formattingProvider` in your LSP config to "brittany" and just run the formatting command. Also the pre-commit hook will format any staged files, so we shouldn't need to format everything ever again.

**Comment by masaeedu** [comment]
Sick, this looks really good. One last thing, can we shove all the nix related stuff (besides shell.nix) in some subfolder?

**Comment by JonathanLorimer** [comment]
@masaeedu yeah, but `flake.nix` and `shell.nix` both need to be top level, so its really only docker.nix that needs to be moved

**Comment by JonathanLorimer** [comment]
@solomon-b isn't happy with the formatter, which is fair. I think we should take a moment to review the options, since its hard to revert these kinds of changes. Options in my eyes are:
- Fourmolu
- stylish-haskell
- brittany

I know, fourmolu supports `ImportQualifiedPost` would have to look into stylish-haskell

**Comment by masaeedu** [comment]
@JonathanLorimer I think (hope?) that we won't actually need to _revert_ anything if we change our minds later, we'll just have to tweak the stuff you've already set up to use a different formatter and then just run it.

As for the folder, can we make it a non-hidden folder? Like, just `nix` instead of `.nix` (hidden folders are annoying when it comes to editing/search tooling).

**Comment by JonathanLorimer** [comment]
oh yeah, you are right @masaeedu switching formatters is pretty trivial. Maybe we should just merge this then.

**Comment by solomon-b** [comment]
+1 on the non-hidden folder for  `.nix`. We can merge after you do change that (unless you have a reason for it to be hidden?)

**Comment by JonathanLorimer** [comment]
@solomon-b @masaeedu alright, made the folder visible

**Comment by solomon-b** [comment]
Thank you @JonathanLorimer

**Pull request merged**